x-pack/plugins/security_solution/common/api/detection_engine/prebuilt_rules/model/diff/three_way_diff/three_way_diff_outcome.ts

```diff
@@ -70,3 +118,29 @@ export const determineIfValueCanUpdate = (diffCase: ThreeWayDiffOutcome): boolea
     diffCase === ThreeWayDiffOutcome.CustomizedValueCanUpdate
   );
 };
+
+/**
+ * Returns a boolean if 2 arrays contain same the elements agnostic of order after being deduplicated
+ *
+ * NOTE: array equality is case insensitive
+ */
+const arraysHaveSameElements = <T>(arr1: T[], arr2: T[]) => {
+  const set1 = new Set(
+    arr1.map((value) => (typeof value === 'string' ? value.toLowerCase() : value))
+  );
+  const set2 = new Set(
+    arr2.map((value) => (typeof value === 'string' ? value.toLowerCase() : value))
+  );
+
+  if (set1.size !== set2.size) {
+    return false;
+  }
+
+  for (const val of set1) {
+    if (!set2.has(val)) {
+      return false;
+    }
+  }
+
+  return true;
```

Comment:
[_.isEqual](https://lodash.com/docs/4.17.15#isEqual) can compare sets:

```suggestion
  return isEqual(set1, set2);
```